Minor changes

notEvil opened the pull request:
Hi,

those are just minor things that would pollute the list of issues and aren't worth individual PRs imo. If you think otherwise, let me know.

- api/network/PHttpServer.serve: changed to parse HTTP POST body

The current code ignores this body and params aren't populated as a consequence.

- api/network/PHttpServer: added method __stop as a quickfix

See https://github.com/victordiaz/PHONK/issues/131
Should receive a proper fix (using interface/base class) in the future

- api/PApp: added method runOnMainLoop

There is `PApp.runOnUiThread` for activities but it doesn't work for services. This works for both

- api/PNetwork: added createHttpServer overload with additional argument ip

In case you want to restrict the server to a subnet, like 127.0.0.1

- utils/AndroidUtils.setWakeLock: fixed bug related to incorrect behavior for some call sequences

The current code would try to release if called with `false` twice. This works with any call sequence


I've tested these changes for a week while developing an app and had no troubles :)

Commit:
- api/network/PHttpServer.serve: changed to parse HTTP POST body
- api/network/PHttpServer: added method __stop as a quickfix
- api/PApp: added method runOnMainLoop
- api/PNetwork: added createHttpServer overload with additional argument ip
- utils/AndroidUtils.setWakeLock: fixed bug related to incorrect behavior for some call sequences

victordiaz reviewed (approved):
Good looks to me. Thanks for the improvements! And please don't hesitate on changing more if you need it :)

Commit:
- app/layout/connection_info_item: fixed linter errors related to txtEventDate is not a direct child of ConstraintLayout
- app/build.gradle: changed to allow debug builds without a signing.properties file
- editor/package.json: changed to create target directory if it doesn't exist
- examples/package.json: changed to create target directory if it doesn't exist

notEvil commented:
Alright, just pushed a few more

- app/layout/connection_info_item: fixed linter errors related to txtEventDate is not a direct child of ConstraintLayout

I'm not familiar with Android development, so you should check if this is reasonable. At least my Android Studio compiles with this change and the Design view looks no different.

- app/build.gradle: changed to allow debug builds without a signing.properties file
- editor/package.json: changed to create target directory if it doesn't exist
- examples/package.json: changed to create target directory if it doesn't exist

Those fixes some pain points when starting from scratch.

In the meantime I realized that the overload of `createHttpServer` doesn't align with the 1 argument version (port shifts to the right). Should I change this?

victordiaz commented:
- The changes you did are perfect, thanks! 
- It makes sense what you say about the ip, port order. But also I think it looks more natural also to say ip, port. I let you pick whatever you think is nice.

Just to let you know, I started this project long ago, so it has lots of suboptimal code decisions. Also I think the javascript world changed a lot, so the API does not feel very modern :) 

I really appreciate every PR and comment. It makes me willing to keep working on it :)

notEvil commented:
> * It makes sense what you say about the ip, port order. But also I think it looks more natural also to say ip, port. I let you pick whatever you think is nice.

Ok, then I won't change it. I've seen some overloads where this happens already and I personally prefer the "natural" sequence. If only there were keyword arguments ... xd

> Just to let you know, I started this project long ago, so it has lots of suboptimal code decisions. Also I think the javascript world changed a lot, so the API does not feel very modern :)

True, but it otherwise works very well, even on my phone (~2017, community rom, android 11). So it passed the test of time imo.

> I really appreciate every PR and comment. It makes me willing to keep working on it :)

Thats great, happy to contribute :)

victordiaz commented:
I will merge this PR then
